superchain/superchain.go

```diff
@@ -171,105 +172,170 @@ func (c *ChainConfig) setNilHardforkTimestampsToDefaultOrZero(s *SuperchainConfi
 	}
 }
 
-// EnhanceYAML creates a customized yaml string from a RollupConfig. After completion,
-// the *yaml.Node pointer can be used with a yaml encoder to write the custom format to file
-func (c *ChainConfig) EnhanceYAML(ctx context.Context, node *yaml.Node) error {
-	hexStringRegex := regexp.MustCompile(`^0x[a-fA-F0-9]+$`)
-
-	// Check if context is done before processing
-	if err := ctx.Err(); err != nil {
-		return fmt.Errorf("context error: %w", err)
+func (c ChainConfig) MarshalTOML() ([]byte, error) {
+	// Uses []outField to deterministically set the order of the toml based on the order of fields
+	// in the ChainConfig struct. Otherwise the fields are ordered alphabetically
+	type outField struct {
+		key   string
+		value interface{}
 	}
```

Comment:
I understand that we can run into problems here if the `ChainConfig` struct ordering has unnested fields listed after nested fields, since the resulting file will have an incorrect information hierarchy.

Can we detect that inside this function and error or panic with a clear warning about how to fix this?

Reply:
Added code to detect this situation. Tested manually by moving some fields around. One tricky part is ignoring `inline` structs. https://github.com/ethereum-optimism/superchain-registry/pull/386/commits/fe0700ae674b787f3ece802bbc223edd71905be1